### PR TITLE
Added sql test code and modified some code of sql.go

### PR DESCRIPTION
**Pull request by bourbonkk**
Added sql test code.

I want to help with your project.
Is this project still going on? It doesn't seem to be active.
I want to make it easy for other users to use by writing a lot of test examples.
And the golang version is 1.14 right?
invite me to slack

### CODE REVIEW

**Comment by jbguerraz** [comment]
Hello @bourbonkk 

Thank you for your contribution :)

If you wanna add a test for sql queries, maybe the best place is here https://github.com/grafadruid/go-druid/blob/master/query.go , actually, in a query_test.go file to add. If you're about examples, then see https://github.com/grafadruid/go-druid/tree/master/examples

It started from go 1.14 but no issue to update that, still, it'll works with most recent golang versions AFAIK.

The project is active and used; it doesn't move so much since it's quite complete and stable, so far so good :)

There are still Druid (community) extensions  to support and tests to add here and there probably.

I've updated the slack invite link, here it is: https://join.slack.com/t/grafadruid/shared_invite/zt-1ohtquqg3-25ka1yV4Kkig0lt5XzfmeA

Cheers,

**Comment by vigith** [comment]
+1, we use go-druid actively in production

**Comment by bourbonkk** [comment]
Thank you for your explanation. 
@jbguerraz 
I'll make the edits you described and commit again. Thank you.

File: builder/query/sql.go [hunk 1]
```diff
@@ -15,8 +15,8 @@ type SQL struct {
 }
 
 type SQLParameter struct {
-	Type  string `json:"type,omitempty"`
-	Value string `json:"value,omitempty"`
+	Type  string      `json:"type,omitempty"`
+	Value interface{} `json:"value,omitempty"`
```
Comment (bourbonkk) [inline builder/query/sql.go:19]: There were cases where the type of value was INTEGER or FLOAT, so I changed it to an interface.

File: builder/query/sql.go [hunk 2]
```diff
@@ -25,8 +25,11 @@ func NewSQL() *SQL {
 	return s
 }
 
-func NewSQLParameter() *SQLParameter {
-	p := &SQLParameter{}
+func NewSQLParameter(Type string, Value interface{}) SQLParameter {
```
Comment (bourbonkk) [inline builder/query/sql.go:28]: Modified to allow use of SQLParameter's constructor.

File: builder/query/sql.go [hunk 3]
```diff
@@ -60,6 +63,11 @@ func (s *SQL) SetParameters(parameters []SQLParameter) *SQL {
 	return s
 }
 
+func (s *SQL) SetContext(context map[string]interface{}) *SQL {
```
Comment (bourbonkk) [inline builder/query/sql.go:66]: I added the SetContext method of the SQL struct because it did not exist.

File: builder/query/query_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
+package query
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestSelectSQL(t *testing.T) {
```
Comment (bourbonkk) [inline builder/query/query_test.go:9]: This is a comparison test of query builder and actual output.

File: builder/query/query_test.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      "value": "2016-06-27T00:00:11.080Z"
+    }
+  ]
+}`
+	expressionJSON, err := json.Marshal(query)
+	assert.Nil(t, err)
+	assert.JSONEq(t, expected, string(expressionJSON))
+}
+
+func TestAllSetterSQL(t *testing.T) {
```
Comment (bourbonkk) [inline builder/query/query_test.go:49]: This is a comparison test when all methods are changed to true

File: examples/querySqlExample.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"github.com/davecgh/go-spew/spew"
+	"github.com/grafadruid/go-druid"
+	"github.com/grafadruid/go-druid/builder/query"
+	"log"
+)
+
+func main() {
+	d, err := druid.NewClient("http://localhost:8888")
```
Comment (bourbonkk) [inline examples/querySqlExample.go:14]: Example of using SQL query

**Comment by raakasf** [comment]
LGTM but I'll get @vigith approve this.

**Comment by jbguerraz** [comment]
LGTM thank you a lot @bourbonkk :)

**Review by jbguerraz — APPROVED** [review]
:fireworks:

**Pull request merged**